### PR TITLE
Ban use of xtra functions that lead to hard-to-spot bugs

### PR DESCRIPTION
**Pull request by thomaseizinger**
Read commit messages for details.

### CODE REVIEW

**Review by da-kami — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Hmm, the integration tests are failing with this change. Weird.

**Comment by da-kami** [comment]
> Hmm, the integration tests are failing with this change. Weird.

It seems that the next order is not received correctly. I think it is the way we wait for the next order initially in the tests.
I was already wondering why we have to await `None` first there - I think we should not have to. (For the cfd list we don't have to wait for an empty vec initially...)

Maybe this change also has something to do with it (but I think the tests are actually failing earlier, on the initial order after sending it out):

```
self.takers
            .send(maker_inc_connections::BroadcastOrder(None))
            .await?;
```

I cannot explain why it behaves like this though 😅

File: daemon/src/maker_cfd.rs [hunk 1]
```diff
@@ -346,17 +349,27 @@ where
...
                 command: TakerCommand::NotifySettlementAccepted { id: order_id },
             })
-            .await?;
+            .await?
+        {
+            Ok(_) => {
+                self.current_agreed_proposals
+                    .insert(order_id, self.get_settlement_proposal(order_id)?);
+                self.remove_pending_proposal(&order_id)
+                    .context("accepted settlement")?;
```
Comment (klochowicz) [inline daemon/src/maker_cfd.rs:364]: nit: this probably could be pulled out of this clause, at it is the same for both cases?

**Review by klochowicz — APPROVED** [review]
nice! looking forward to testing this more :)

File: daemon/src/maker_cfd.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/maker_cfd.rs:364]: True!

**Comment by thomaseizinger** [comment]
bors retry

**Pull request merged**